### PR TITLE
Tutorial transpose reshape

### PR DESCRIPTION
**Pull request by NRauschmayr**
## Description ##
(Brief description on what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: docs/tutorials/basic/reshape_transpose.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+The color image has the following properties:
+    - width: 210 pixels
+    - height: 200 pixels
+    - colors: 3 (RGB)
+
+Now lets reshape the image in order to exchange width and height dimension.
+
+
+```python
+reshaped = img_array.reshape(200,210,3)
```
Comment (ThomasDelteil) [inline docs/tutorials/basic/reshape_transpose.md]: use `.reshape((200, 210, 3))`

File: docs/tutorials/basic/reshape_transpose.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    (200, 210, 3)
+
+![png](https://raw.githubusercontent.com/NRauschmayr/web-data/tutorial_transpose_reshape/mxnet/doc/tutorials/basic/transpose_reshape/reshaped_image.png) <!--notebook-skip-line-->
+
+
+As we can see the first and second dimensions have changed. However the image can't be identified as cat anylonger. In order to understand what happened, let's have a look at the image below.
+
+<img src="https://raw.githubusercontent.com/NRauschmayr/web-data/tutorial_transpose_reshape/mxnet/doc/tutorials/basic/transpose_reshape/reshape.png" style="width:700px;height:300px;">
+
+While the number of rows and columns changed, the layout of the underlying data did not. The pixel values that have been in one row are still in one row. This means for instance that pixel 10 in the upper right corder ends up in the middle of the image instead of the upper left corner. Consequently contextual information gets lost, because the relative position of pixel values is not the same anymore. As one can imagine a neural network would not be able to classify such an image as cat. `Transpose` instead changes both: the dimensions but also the corresponding pixel values.
```
Comment (ThomasDelteil) [inline docs/tutorials/basic/reshape_transpose.md]: "`Transpose` instead changes both: the dimensions but also the corresponding pixel values." not sure what you mean here

File: docs/tutorials/basic/reshape_transpose.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+
+```python
+transposed = img_array.transpose(1,0,2)
```
Comment (ThomasDelteil) [inline docs/tutorials/basic/reshape_transpose.md]: `.transpose((1, 0, 2))`

File: docs/tutorials/basic/reshape_transpose.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+```
+
+![png](https://raw.githubusercontent.com/NRauschmayr/web-data/tutorial_transpose_reshape/mxnet/doc/tutorials/basic/transpose_reshape/transposed_image.png) <!--notebook-skip-line-->
+
+
+As we can see width and height changed, by rotating pixel values by 90 degrees. Transpose does the following:
+
+<img src="https://raw.githubusercontent.com/NRauschmayr/web-data/tutorial_transpose_reshape/mxnet/doc/tutorials/basic/transpose_reshape/transpose.png" style="width:700px;height:300px;">
+
+As shown in the diagram, pixel values that have been in the first row are now in the first column.
```
Comment (ThomasDelteil) [inline docs/tutorials/basic/reshape_transpose.md]: add: The axis have been flipped

File: docs/tutorials/basic/reshape_transpose.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+## When to transpose with MXNet
+In this chapter we discuss when transpose and reshape is used in MXNet. 
+#### Channel first for images
+Images are usually stored in the format height, wight, channel. When working with convolutional layers, MXNet expects the layout to be `NCHW` (batch, channel, height, width). Consequently, images need to be transposed to have the right format. For instance, you may have a function like the following:
```
Comment (ThomasDelteil) [inline docs/tutorials/basic/reshape_transpose.md]: add that's it done for performance

File: docs/tutorials/basic/reshape_transpose.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+``` 
+def transform(data, label): 
+     return data.astype(np.float32).transpose((2,0,1))/255.0, float(label)
```
Comment (ThomasDelteil) [inline docs/tutorials/basic/reshape_transpose.md]: change `float(label)` to `np.float32(label)`

File: docs/tutorials/basic/reshape_transpose.md [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+```
+Images may also be stored as 1 dimensional vector, e.g. instead of `[28,28,1]` you may have `[784,1]`. In this situation you need to perform a reshape e.g. `ndarray.reshape((1,28,28))`
```
Comment (ThomasDelteil) [inline docs/tutorials/basic/reshape_transpose.md]: for example in byte packed datasets

File: docs/tutorials/basic/reshape_transpose.md [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+#### TNC layout for RNN
+When working with LSTM or GRU layers, the default layout for input and ouput tensors has to be `TNC` (sequence length, batch size, and feature dimensions). For instance in the following network the input goes into a 1 dimensional convolution layer and whose output goes into a GRU cell. Here the tensors would mismatch, because `Conv1D` takes data as `NCT`, but GRU  expects it to be `NTC`. 
+```
+class model(mx.gluon.nn.Block):
+    def __init__(self):
+        super(model, self).__init__()
+        
+        self.sequential1 = mx.gluon.nn.Sequential()
+        self.sequential2 = mx.gluon.nn.Sequential()
+        self.sequential1 = mx.gluon.nn.Sequential()
```
Comment (ThomasDelteil) [inline docs/tutorials/basic/reshape_transpose.md]: create them under the name scope and you have created them twice

File: docs/tutorials/basic/reshape_transpose.md [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+        self.sequential2 = mx.gluon.nn.Sequential()
+        with self.name_scope():    
+            self.sequential1.add(mx.gluon.nn.Conv1D(196, kernel_size=15, strides=4)) 
+            self.sequential1.add(mx.gluon.nn.BatchNorm(axis=1))
+            self.sequential1.add(mx.gluon.nn.Activation(activation='relu')) 
+            self.sequential1.add(mx.gluon.nn.Dropout(0.8)) 
+
+            self.sequential2.add(mx.gluon.rnn.GRU(128, layout='NTC'))
```
Comment (ThomasDelteil) [inline docs/tutorials/basic/reshape_transpose.md]:
```
network = gluon.nn.HybridSequential()
with network.name_scope():
   network.add(gluon.nn.Conv1D(....))
   network.add(gluon.nn.HybridLambda(lambda F, x: F.transpose(x, (0, 2, 1)))
   network.add(gluon.rnn.GRU(128)
```

File: docs/tutorials/gluon/Untitled.ipynb [hunk 10]
```diff
@@ -0,0 +1,6 @@
+{
+ "cells": [],
+ "metadata": {},
+ "nbformat": 4,
```
Comment (ThomasDelteil) [inline docs/tutorials/gluon/Untitled.ipynb:4]: welcome back untitled

File: docs/tutorials/index.md [hunk 11]
```diff
@@ -57,7 +57,7 @@ Select API:&nbsp;
     * [Logistic Regression](/tutorials/gluon/logistic_regression_explained.html)
     * [Word-level text generation with RNN, LSTM and GRU](http://gluon.mxnet.io/chapter05_recurrent-neural-networks/rnns-gluon.html) <img src="https://upload.wikimedia.org/wikipedia/commons/6/6a/External_link_font_awesome.svg" alt="External link" height="15px" style="margin: 0px 0px 3px 3px;"/>
     * [Visual Question Answering](http://gluon.mxnet.io/chapter08_computer-vision/visual-question-answer.html) <img src="https://upload.wikimedia.org/wikipedia/commons/6/6a/External_link_font_awesome.svg" alt="External link" height="15px" style="margin: 0px 0px 3px 3px;"/>
-   * [Image similiarity search with InfoGAN](/docs/tutorials/gluon/info_gan.md)<img src="https://upload.wikimedia.org/wikipedia/commons/6/6a/External_link_font_awesome.svg" alt="External link" height="15px" style="margin: 0px 0px 3px 3px;"/>
+   * [Image similiarity search with InfoGAN](/tutorials/gluon/info_gan.html)
```
Comment (ThomasDelteil) [inline docs/tutorials/index.md:60]: info gan?

File: docs/tutorials/gluon/info_gan.md [hunk 12]
```diff
@@ -171,7 +171,7 @@ class Discriminator(gluon.HybridBlock):
 ```
 
 The InfoGAN has the following layout.
-<img src="https://raw.githubusercontent.com/NRauschmayr/web-data/master/mxnet/doc/tutorials/info_gan/InfoGAN.png" style="width:800px;height:250px;">
+<img src="https://raw.githubusercontent.com/dmlc/web-data/master/mxnet/doc/tutorials/info_gan/InfoGAN.png" style="width:800px;height:250px;">
 Discriminator and Generator are the same as in the DCGAN example. On top of the Disciminator is the Q model, which is estimating the code `c` for given fake images. The Generator's input is random noise and the latent code `c`.
```
Comment (ThomasDelteil) [inline docs/tutorials/gluon/info_gan.md:175]: remove info gan commits

**Review by ThomasDelteil — CHANGES_REQUESTED** [review]
